src/snowflake/snowpark/mock/_functions.py

```diff
@@ -1085,6 +1085,43 @@ def mock_to_variant(expr: ColumnEmulator):
     return res
 
 
+def _object_construct(exprs, drop_nulls):
+    import pandas
+
+    expr_count = len(exprs)
+    if expr_count % 2 != 0:
+        raise TypeError(
+            f"Cannot construct an object from an odd number ({expr_count}) of values."
+        )
+
+    if expr_count == 0:
+        return ColumnEmulator(data=[dict()])
+
+    def construct_dict(x):
+        return {
+            x[i]: x[i + 1]
+            for i in range(0, expr_count, 2)
+            if x[i] is not None and not (drop_nulls and x[i + 1] is None)
```

Comment:
NIT: Just a heads up that [documentation](https://docs.snowflake.com/en/sql-reference/functions/object_construct#usage-notes) suggests SQL null and JSON null are treated differently here. I'm not sure if Local Testing distinguishes the two.